app.js

```diff
@@ -21,7 +22,28 @@ app.all('/*', (request, response, next) => {
 	response.status(404).send({ msg: 'not found' });
 });
 
-// error handling
+// register error handling middleware: pSQL errors
+app.use((err, request, response, next) => {
+	/**
+	 * 22P02: invalid text representation
+	 * 42601: syntax error
+	 * 42703: undefined column or parameter (limit is cast to NaN << throws this code)
+	 * 23502: violates not null constraint
+	 */
+	const badRequestErrCodes = ['22P02', '42601', '42703', '23502'];
+	if (badRequestErrCodes.includes(err.code)) {
+		response.status(400).send({ msg: 'bad request' });
+	}
+	next(err);
+});
```

Comment:
I like this error handler, however, you don't need to add all the codes yet. Only write enough code to pass the test you are on. The other codes can be added when they are being tested for whenever they're needed. Please do delete any code that are not needed just yet